[diff2html] feat add syntax highlighting

campersau opened the pull request:
Currently it uses the github colors but they aren't optimised for ungit yet. Maybe someone else can help.
This does not need to go into the diff2html PR. I just wanted to show how it would look like.

![image](https://cloud.githubusercontent.com/assets/4009570/7892438/b20342ee-0654-11e5-964a-396d1bef428e.png)


fixes FredrikNoren/ungit#500

codingtwinky commented:
I had something similar before but the problem was that "+" and "-" gave some hard time in some languages.

And I think this would be a great addition to diff2html so I think creating this against diff2html would be great for all of it's users.  Maybe a config to toggle.

campersau commented:
Updated some styles (they are now based on the dark atom ide theme), added config toggle and fixed the highlighting for `+`, `-`.

![image](https://cloud.githubusercontent.com/assets/4009570/7893155/36e48fe4-065b-11e5-925e-2362640b8d9c.png)

codingtwinky commented:
Also, can we talk about possibility of this feature being pushed to diff2html rather then ungit?  I think it belongs there and we can do that more efficiently within diff2html don't you think?

I'm sure other users of diff2html would love to have this...
